code/modules/reagents/chemistry/reagents.dm

```diff
@@ -63,7 +63,7 @@
 
 	var/mob/living/carbon/C = M
 	if(method == REAGENT_INGEST && istype(C) && C.get_blood_id() == id)
-		if(id == "blood" && !(data?["blood_type"] in get_safe_blood(C.dna?.blood_type)))
+		if(id == "blood" && !(data?["blood_type"] in get_safe_blood(C.dna?.blood_type))|| C.dna?.species.own_species_blood && C.dna?.species.name != data?["species"] || data?["species_only"] && C.dna?.species.name != data?["species"])
```

Comment:
It probably couldn't hurt to move this check into a proc since you reuse it a few times

Reply:
Not sure if I can proc it due to how it checks it, but will see